lib/helpers/populate/getModelsMapForPopulate.js

```diff
@@ -101,13 +102,16 @@ module.exports = function getModelsMapForPopulate(model, docs, options) {
       } else {
         localField = virtualPrefix + virtual.options.localField;
       }
+      if (typeof virtual.options.foreignField === 'function') {
+        foreignField = virtualPrefix + virtual.options.foreignField.call(doc, doc);
```

Comment:
Looks like there's a bunch of test failures due to this change. I think they should be fixed if you get rid of the `virtualPrefix +` bit here. Should be just `foreignField = virtual.options.foreignField.call(doc, doc)` I think.